File: Java/AbstractFactory/FurnitureStore/src/com/laxmena/ClientDemo.java
package com.laxmena;

import com.laxmena.chair.Chair;
import com.laxmena.factory.ArtDecoFurnitureFactory;
import com.laxmena.factory.FurnitureFactory;
import com.laxmena.factory.ModernFurnitureFactory;
import com.laxmena.factory.VictorianFurnitureFactory;
import com.laxmena.sofa.Sofa;
import com.laxmena.table.Table;

public class ClientDemo {

    public static void main(String[] args) {
        FurnitureFactory factory = null;

        System.out.println("Abstract Factory Demo\n\n");

	    // Client 1 requests for Victorian Furniture's - Chair and a Table
        System.out.println("User1 is Interested in Victorian Style");
        factory = new VictorianFurnitureFactory();
        Chair user1Chair = factory.createChair();
        Table user1Table = factory.createTable();
        user1Chair.sitOn();
        System.out.println("Weight capacity of User1 Table: " + user1Table.weightCapacity());

        // Client 2 requests for ArtDeco Furniture's
        System.out.println("\nUser2 is Interested in ArtDeco Style");
        factory = new ArtDecoFurnitureFactory();
        Sofa user2Sofa = factory.createSofa();
        Table user2Table = factory.createTable();
        System.out.println("Sofa material: " + user2Sofa.cushionType());
        System.out.println("Table weight capacity: " + user2Table.weightCapacity());

        // Client3 requests all furnitures in modern style
        System.out.println("\nUser3 is fond of Modern Style!");
        factory = new ModernFurnitureFactory();
        Chair user3chair = factory.createChair();
        Sofa user3sofa = factory.createSofa();
        Table user3table = factory.createTable();
        System.out.println("User3 Chair has legs?: legs count " + user3chair.legsCount());
        System.out.println("Sofa with cushion? " + user3sofa.hasCushion());
        System.out.println("Table material: " + user3table.material());

    }
}

File: AbstractFactory/Python/FurnitureStore/client_demo.py
from factory.art_deco_furniture_factory import ArtDecoFurnitureFactory
from factory.furniture_factory import FurnitureFactory
from factory.mordern_furniture_factory import ModernFurnitureFactory
from factory.victorian_furniture_factory import VictorianFurnitureFactory
from products.chair import Chair
from products.sofa import Sofa
from products.table import Table

if __name__ == '__main__':
    print("Abstract Factory Demo\n")

    # Victorian Style
    factory: FurnitureFactory = VictorianFurnitureFactory()
    chair1: Chair = factory.create_chair()
    table1: Table = factory.create_table()
    chair1.sit_on()
    print(table1.weight_capacity())

    # ArtDeco Style
    factory = ArtDecoFurnitureFactory()
    chair2: Chair = factory.create_chair()
    sofa2: Sofa = factory.create_sofa()
    print("Legs count: ", chair2.legs_count())
    print("Cushion Type: ", sofa2.cushion_type())

    # Modern Style
    factory = ModernFurnitureFactory()
    table3: Table = factory.create_table()
    sofa3: Sofa = factory.create_sofa()
    print("Table load capacity: ", table3.weight_capacity())
    print("Modern Cushion Type? ", sofa3.cushion_type())
